src/metricscollector/collector/apppoller_test.go

```diff
@@ -118,19 +141,33 @@ var _ = Describe("Apppoller", func() {
 				})
 
 				It("saves non-empty metrics to database", func() {
-					Expect(database.SaveMetricCallCount()).To(Equal(2))
+					fclock.Increment(TestPollInterval * time.Second)
+					Eventually(database.SaveMetricCallCount).Should(Equal(1))
+
+					fclock.Increment(TestPollInterval * time.Second)
+					Consistently(database.SaveMetricCallCount).Should(Equal(1))
+
+					fclock.Increment(TestPollInterval * time.Second)
+					Eventually(database.SaveMetricCallCount).Should(Equal(2))
 				})
 			})
 		})
 
 		Context("when retrieving container metrics all fails", func() {
 
 			BeforeEach(func() {
-				noaa.ContainerMetricsReturns(nil, errors.New("an error"))
+				noaa.ContainerMetricsReturns(nil, errors.New("test apppoller error"))
 			})
 
-			It("saves nothing to database", func() {
-				Expect(database.SaveMetricCallCount()).To(Equal(0))
+			It("saves nothing to database and logs the errors", func() {
+				fclock.Increment(TestPollInterval * time.Second)
+				Consistently(database.SaveMetricCallCount).Should(BeZero())
+				Eventually(buffer).Should(gbytes.Say("test apppoller error"))
```

Comment:
Checking for the error is insufficient since you have 3 places that could log it. You need to look for poll-metric-from-noaa.
At which point the consistently could be an Expect instead.